### PR TITLE
Fix LottieAnimationView.setProgress called from the view init method prevents the progress from being restored from the saved state

### PR DESCRIPTION
**Pull request by flaringapp**
See [#2069](https://github.com/airbnb/lottie-android/issues/2069)

### CODE REVIEW

**Comment by gpeal** [comment]
If you were to set an initial progress but then were to subsequently change it or play the animation, wouldn't restoration overwrite the saved progress with what is set in XML?

**Comment by flaringapp** [comment]
@gpeal I think it makes sense to check whether there is a progress attribute set in the XML and if yes - treat it as user-provided and prevent saved state restoration from overwriting the progress value.

Also it will be nice to provide an option to restore saved progress even if the progress is set in the XML. I suggest adding an attribute named `lottie_useRestoredProgress` with default value `false`. So if user wants to explicitly restore progress state instead of starting the animation from the specified progress, he can use this attribute. Behind the scenes it'll affect the `setProgressInternal` called from attributes init method. If the attribute is `true`, then we won't save the `UserActionTaken.SET_PROGRESS` flag (`fromUser = false`) or do exactly the opposite with default `false` value

**Comment by gpeal** [comment]
@flaringapp I'm still trying to wrap my head around the exact use case that's failing here. Could you create a sample project that highlights the issue?

**Comment by flaringapp** [comment]
@gpeal Please take a look at this [sample project](https://github.com/flaringapp/LottieIssue)
In scope of this issue I expect the animation to persist it's previous progress when a fragment is restored

File: lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java [hunk 1]
```diff
@@ -977,7 +980,15 @@ public int getFrame() {
   }
 
   public void setProgress(@FloatRange(from = 0f, to = 1f) float progress) {
-    userActionsTaken.add(UserActionTaken.SET_PROGRESS);
+    setProgressInternal(true, progress);
+  }
+
+  private void setProgressInternal(
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java:986]: Could you swap the parameter order here?

**Review by gpeal — COMMENTED** [review]
Okay thanks for the sample. I've been very busy lately but found some time to take a look at this. let's move forward with your implementation as-is. Just swap the parameter order here but other than that, LGTM. Thanks for doing this.

File: lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java (hunk 1 above)
Comment (flaringapp) [reply lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java:986]: @gpeal Done

Comment (flaringapp) [reply lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java:986]: I've rebased on master as well

**Review by gpeal — APPROVED** [review]

**Pull request merged**